### DISCUSSION TITLE
Managing self hosted CA using terraform

### DISCUSSION THREAD

**Discussion by slavb18** [smallstep/certificates#1722] [Q&A]
Hi!

Does it possible to manage self hosted CA with remote provisioners management enabled using [terraform](https://registry.terraform.io/providers/smallstep/smallstep/latest/docs/resources/provisioner)  ?

**Comment by tashian** [comment]
You'd have to write a provider to do it. We have some Puppet and Ansible automations in our [community-contributed integrations list](https://github.com/smallstep/certificates/discussions/765), but not Terraform.

Alternatively, our [commercial CA](https://smallstep.com/docs/platform/) has a [Terraform provider](https://github.com/smallstep/terraform-provider-smallstep) that includes [authority](https://registry.terraform.io/providers/smallstep/smallstep/latest/docs/resources/authority) and [provisioner management](https://registry.terraform.io/providers/smallstep/smallstep/latest/docs/resources/provisioner)
